file_browser/src/components/AddDIalog.tsx

```diff
@@ -0,0 +1,112 @@
+import { useState } from "react";
+import { SystemItem } from "../classes/SystemItem";
+import { FileClass } from "../classes/File";
+import { Folder } from "../classes/Folder";
+import { observer } from "mobx-react-lite";
+import fileSystemStore from "../stores/FileSystemStore";
+
+const options = ["File", "Folder"];
+
+const AddDialog = observer(({
+  onClose
+}: {
+  onClose: (state: boolean) => void;
+}) => {
+
+  enum Selected {
+    FileOption = 0,
+    FolderOption = 1
+  }
+
+  const [selected, setSelected] = useState<number>(Selected.FileOption);
+  const [name, setName] = useState("");
+  const [content, setContent] = useState("");
+  const [error, setError] = useState("");
+
+  const handleAdd = (item: SystemItem) => {
+    (fileSystemStore.selectedItem as Folder).add(item);
+  };
```

Comment:
On contray of my other comment, this one doesn't actually handle an event, but rather performs an action. In this case, you can call it just `add` or `addToFolder` to add more context to the reader.